Lib/test/test_random.py

```diff
@@ -88,6 +89,25 @@ def test_shuffle(self):
         self.assertTrue(lst != shuffled_lst)
         shuffle(lst)
         self.assertTrue(lst != shuffled_lst)
+        self.assertRaises(TypeError, shuffle, (1, 2, 3))
+        self.assertRaises(TypeError, shuffle, {1, 2, 3})
+        self.assertRaises(TypeError, shuffle, 'shuffle')
+        self.assertRaises(KeyError, shuffle, dict(a=1, b=2))
+
+    @unittest.expectedFailure
+    def test_shuffle_dict_with_numeric_keys(self):
+        shuffle = self.gen.shuffle
+        seq = dict(zip(range(9), 'abcdefghi'))
+        self.assertRaises(KeyError, shuffle, seq)
```

Comment:
I think this isn't a valid test and should be dropped.  It is an effect incidental to the implementation (due to duck-typing).